test/utils/TestUtils.t.sol

```diff
@@ -73,7 +73,18 @@ contract TestUtils is Test {
         return asset_.getDigest(keccak256(abi.encode(asset_.CANCEL_AUTHORIZATION_TYPEHASH(), from_, fromNonce_)));
     }
 
+    function _getVS(uint8 v_, bytes32 s_) internal pure returns (bytes32 vs_) {
+        if (v_ == 28) {
+            // then left-most bit of s has to be flipped to 1 to get vs
+            vs_ = s_ | bytes32(uint256(1) << 255);
+        }
```

Comment:
What is `vs_` if `v_ != 28`?

Reply:
Ah yes, we should return `s_` here. Otherwise is means that `vs_` would be `bytes32(0)`

If `v_ == 28`, it means that the `v` parameter equals `1`, so we need to flip the left most bit of `s_` to `1`.
Otherwise, it  means that `v` is `0` and we can leave the left most bit of `s_` to `0`.

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/8/commits/b08fe3cd46db0bcd272a1ce76470bca0d5089713